components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/NetworkMessageEncoder.cs

```diff
@@ -366,10 +380,7 @@ public class NetworkMessageEncoder : IMessageEncoder {
                             .Select(q => q.Any() ? q.Dequeue() : null)
                                 .Where(s => s != null)
                                     .ToDictionary(
-                                        s => !string.IsNullOrEmpty(s.Id)
-                                            ? s.Id
-                                            : s.NodeId.ToExpandedNodeId(context.NamespaceUris)
-                                                .AsString(message.ServiceMessageContext),
+                                        s => GetPayloadIdentifier(s, message,context),
```

Comment:
I had a different approach here, but same like this it was a workaround. If this works properly in all scenarios. I'm happy with it. Did you test the behavior when the publisher is started with the argument --fd (fetch Display Name) on?

Reply:
yes I tested with and without argument --fd=true